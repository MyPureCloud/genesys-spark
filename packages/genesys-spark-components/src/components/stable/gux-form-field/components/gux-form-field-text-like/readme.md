# gux-form-field-text-like



<!-- Auto Generated Below -->


## Properties

| Property        | Attribute        | Description | Type                                    | Default     |
| --------------- | ---------------- | ----------- | --------------------------------------- | ----------- |
| `clearable`     | `clearable`      |             | `boolean`                               | `undefined` |
| `labelPosition` | `label-position` |             | `"above" \| "beside" \| "screenreader"` | `undefined` |
| `loading`       | `loading`        |             | `boolean`                               | `false`     |


## Methods

### `guxForceUpdate() => Promise<void>`



#### Returns

Type: `Promise<void>`




## Slots

| Slot           | Description                     |
| -------------- | ------------------------------- |
| `"error"`      | Optional slot for error message |
| `"help"`       | Optional slot for help message  |
| `"input"`      | Required slot for input tag     |
| `"label"`      | Required slot for label tag     |
| `"label-info"` | Optional slot for label tooltip |
| `"prefix"`     | Optional slot for prefix        |
| `"suffix"`     | Optional slot for suffix        |


## Dependencies

### Used by

 - [gux-pagination-buttons-legacy](../../../../legacy/gux-pagination-legacy/gux-pagination-buttons-legacy)

### Depends on

- [gux-radial-loading](../../../gux-radial-loading)
- [gux-form-field-label-indicator](../../helper-components/gux-form-field-optional-indicator)
- [gux-form-field-input-clear-button](../../helper-components/gux-form-field-input-clear-button)
- [gux-icon](../../../gux-icon)

### Graph
```mermaid
graph TD;
  gux-form-field-text-like --> gux-radial-loading
  gux-form-field-text-like --> gux-form-field-label-indicator
  gux-form-field-text-like --> gux-form-field-input-clear-button
  gux-form-field-text-like --> gux-icon
  gux-form-field-input-clear-button --> gux-icon
  gux-pagination-buttons-legacy --> gux-form-field-text-like
  style gux-form-field-text-like fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
