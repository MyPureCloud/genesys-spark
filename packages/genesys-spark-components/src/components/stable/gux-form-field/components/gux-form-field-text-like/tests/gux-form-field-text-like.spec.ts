import { newSpecPage } from '@test/specTestUtils';

import { GuxFormFieldTextLike } from '../gux-form-field-text-like';

const components = [GuxFormFieldTextLike];
const language = 'en';

describe('gux-form-field-text-like', () => {
  it('should build', async () => {
    const html = `
      <gux-form-field-text-like>
        <input slot="input" type="text" value="Sample text"/>
        <label slot="label">Label</label>
        <span slot="error">Error message</span>
      </gux-form-field-text-like>
    `;
    const page = await newSpecPage({ components, html, language });

    expect(page.rootInstance).toBeInstanceOf(GuxFormFieldTextLike);
  });

  describe('#render', () => {
    describe('clearable', () => {
      [
        { componentAttribute: '', inputAttribute: '' },
        { componentAttribute: 'clearable', inputAttribute: '' },
        { componentAttribute: 'clearable="true"', inputAttribute: '' },
        { componentAttribute: 'clearable="false"', inputAttribute: '' },
        { componentAttribute: 'clearable', inputAttribute: 'disabled' }
      ].forEach(({ componentAttribute, inputAttribute }, index) => {
        it(`should render component as expected (${index + 1})`, async () => {
          const html = `
            <gux-form-field-text-like ${componentAttribute}>
              <input slot="input" type="text" value="Sample text" ${inputAttribute}/>
              <label slot="label">Label</label>
            </gux-form-field-text-like>
          `;
          const page = await newSpecPage({ components, html, language });

          expect(page.root).toMatchSnapshot();
        });
      });
    });

    describe('prefix', () => {
      it('should render component as expected', async () => {
        const html = `
          <gux-form-field-text-like>
            <span slot="prefix">Prefix</span>
            <input slot="input" type="text" value="Sample text"/>
            <label slot="label">Label</label>
          </gux-form-field-text-like>
        `;
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });

    describe('suffix', () => {
      it('should render component as expected', async () => {
        const html = `
          <gux-form-field-text-like>
            <input slot="input" type="text" value="Sample text"/>
            <span slot="suffix">Suffix</span>
            <label slot="label">Label</label>
          </gux-form-field-text-like>
        `;
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });

    describe('loading', () => {
      [
        { componentAttribute: '', inputAttribute: '' },
        { componentAttribute: 'loading', inputAttribute: '' },
        { componentAttribute: 'loading="true"', inputAttribute: '' },
        { componentAttribute: 'loading="false"', inputAttribute: '' },
        { componentAttribute: 'loading', inputAttribute: 'disabled' }
      ].forEach(({ componentAttribute, inputAttribute }, index) => {
        it(`should render component as expected (${index + 1})`, async () => {
          const html = `
            <gux-form-field-text-like ${componentAttribute}>
              <input slot="input" type="text" value="Sample text" ${inputAttribute}/>
              <label slot="label">Label</label>
            </gux-form-field-text-like>
          `;
          const page = await newSpecPage({ components, html, language });

          expect(page.root).toMatchSnapshot();
        });
      });
    });

    describe('help', () => {
      it('should render component as expected', async () => {
        const html = `
          <gux-form-field-text-like>
          <input slot="input" type="text" name="e-1" />
          <label slot="label">Default</label>
          <span slot="help">This is a help message</span>
        </gux-form-field-text-like>
        `;
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });

    describe('label-info', () => {
      it('should render component as expected', async () => {
        const html = `
          <gux-form-field-text-like>
          <input slot="input" type="text" name="e-1" />
          <label slot="label">Default</label>
          <gux-label-info-beta slot="label-info">
            <span slot="content">This is some tooltip text</span>
          </gux-label-info-beta>
        </gux-form-field-text-like>
        `;
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });

    describe('label-position', () => {
      [
        '',
        'label-position="above"',
        'label-position="beside"',
        'label-position="screenreader"'
      ].forEach((componentAttribute, index) => {
        it(`should render component as expected (${index + 1})`, async () => {
          const html = `
            <gux-form-field-text-like ${componentAttribute}>
              <input slot="input" type="text" value="Sample text"/>
              <label slot="label">Label</label>
            </gux-form-field-text-like>
          `;
          const page = await newSpecPage({ components, html, language });

          expect(page.root).toMatchSnapshot();
        });
      });
    });

    describe('input type attribute', () => {
      ['email', 'password', 'text'].forEach((inputTypeAttribute, index) => {
        it(`should render component as expected (${index + 1})`, async () => {
          const html = `
            <gux-form-field-text-like>
              <input slot="input" type="${inputTypeAttribute}" value="Sample text" />
              <label slot="label">Label</label>
            </gux-form-field-text-like>
          `;
          const page = await newSpecPage({ components, html, language });

          expect(page.root).toMatchSnapshot();
        });
      });
    });

    describe('input attributes', () => {
      ['', 'disabled', 'required'].forEach((inputAttribute, index) => {
        it(`should render component as expected (${index + 1})`, async () => {
          const html = `
            <gux-form-field-text-like>
              <input slot="input" type="text" value="Sample text" ${inputAttribute}/>
              <label slot="label">Label</label>
            </gux-form-field-text-like>
          `;
          const page = await newSpecPage({ components, html, language });

          expect(page.root).toMatchSnapshot();
        });
      });
    });

    describe('indicator marks', () => {
      ['', 'indicator-mark="optional"', 'indicator-mark="required"'].forEach(
        (indicatorMark, index) => {
          it(`should render component as expected (${index + 1})`, async () => {
            const html = `
              <gux-form-field-text-like ${indicatorMark}>
                <input slot="input" type="text" value="Sample text"/>
                <label slot="label">Label</label>
              </gux-form-field-text-like>
            `;
            const page = await newSpecPage({ components, html, language });

            expect(page.root).toMatchSnapshot();
          });
        }
      );
    });
  });
});
