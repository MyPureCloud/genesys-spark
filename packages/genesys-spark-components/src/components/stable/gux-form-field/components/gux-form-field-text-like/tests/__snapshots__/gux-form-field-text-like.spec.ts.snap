// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-form-field-text-like #render clearable should render component as expected (1) 1`] = `
<gux-form-field-text-like>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render clearable should render component as expected (2) 1`] = `
<gux-form-field-text-like clearable="">
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
            <gux-form-field-input-clear-button></gux-form-field-input-clear-button>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render clearable should render component as expected (3) 1`] = `
<gux-form-field-text-like clearable="true">
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
            <gux-form-field-input-clear-button></gux-form-field-input-clear-button>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render clearable should render component as expected (4) 1`] = `
<gux-form-field-text-like clearable="false">
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render clearable should render component as expected (5) 1`] = `
<gux-form-field-text-like clearable="">
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-disabled gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input disabled="" id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render help should render component as expected 1`] = `
<gux-form-field-text-like>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help gux-show">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input aria-describedby="gux-form-field-help-i" id="gux-form-field-input-i" name="e-1" slot="input" type="text">
  <label for="gux-form-field-input-i" slot="label">
    Default
  </label>
  <span id="gux-form-field-help-i" slot="help">
    This is a help message
  </span>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render input attributes should render component as expected (1) 1`] = `
<gux-form-field-text-like>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render input attributes should render component as expected (2) 1`] = `
<gux-form-field-text-like>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-disabled gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input disabled="" id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render input attributes should render component as expected (3) 1`] = `
<gux-form-field-text-like>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label gux-required">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" required="" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render input type attribute should render component as expected (1) 1`] = `
<gux-form-field-text-like>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="email" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render input type attribute should render component as expected (2) 1`] = `
<gux-form-field-text-like>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="password" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render input type attribute should render component as expected (3) 1`] = `
<gux-form-field-text-like>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render label-position should render component as expected (1) 1`] = `
<gux-form-field-text-like>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render label-position should render component as expected (2) 1`] = `
<gux-form-field-text-like label-position="above">
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render label-position should render component as expected (3) 1`] = `
<gux-form-field-text-like label-position="beside">
  <mock:shadow-root>
    <div class="gux-beside gux-form-field-container">
      <div class="gux-beside gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render label-position should render component as expected (4) 1`] = `
<gux-form-field-text-like label-position="screenreader">
  <mock:shadow-root>
    <div class="gux-form-field-container gux-screenreader">
      <div class="gux-form-field-label gux-screenreader">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render loading should render component as expected (1) 1`] = `
<gux-form-field-text-like>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render loading should render component as expected (2) 1`] = `
<gux-form-field-text-like loading="">
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <gux-radial-loading context="input"></gux-radial-loading>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render loading should render component as expected (3) 1`] = `
<gux-form-field-text-like loading="true">
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <gux-radial-loading context="input"></gux-radial-loading>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render loading should render component as expected (4) 1`] = `
<gux-form-field-text-like loading="false">
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render loading should render component as expected (5) 1`] = `
<gux-form-field-text-like loading="">
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-disabled gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <gux-radial-loading context="input"></gux-radial-loading>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input disabled="" id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render prefix should render component as expected 1`] = `
<gux-form-field-text-like>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-has-prefix gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <span id="gux-prefix-i" slot="prefix">
    Prefix
  </span>
  <input aria-describedby="gux-prefix-i" id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;

exports[`gux-form-field-text-like #render suffix should render component as expected 1`] = `
<gux-form-field-text-like>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-has-suffix gux-input-container">
            <slot name="prefix"></slot>
            <slot name="input"></slot>
            <slot name="suffix"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input aria-describedby="gux-suffix-i" id="gux-form-field-input-i" slot="input" type="text" value="Sample text">
  <span id="gux-suffix-i" slot="suffix">
    Suffix
  </span>
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-text-like>
`;
